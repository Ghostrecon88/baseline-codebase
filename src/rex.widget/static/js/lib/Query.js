/**
 * Bindings to HTSQL query HTTP API.
 *
 * @copyright 2015, Prometheus Research LLC
 */

export {Query as default} from './data/Query';
