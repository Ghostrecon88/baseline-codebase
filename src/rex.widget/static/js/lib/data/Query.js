/**
 * @copyright 2015, Prometheus Research, LLC
 */

import {fetch} from '../fetch';
import shallowEquals from '../shallowEquals';

export class Query {

  static fetch = fetch;

  constructor(path, params) {
    this.path = path;
    this._params = params;
  }

  params(params) {
    return new this.constructor(this.path, {...this._params, ...params});
  }

  produce(params) {
    let query = {
      ...this._params,
      ...params
    };
    return this.constructor.fetch(this.path, query);
  }

  produceCollection(params) {
    return this.produce(params);
  }

  produceEntity(params) {
    return this.produce(params);
  }

  equals(other) {
    return (
      other &&
      this.constructor === other.constructor &&
      this.path === other.path &&
      shallowEquals(this._params, other._params)
    );
  }
}

export function isQuery(obj) {
  return obj instanceof Query;
}

export default function query(path, params = {}) {
  return new Query(path, params);
}
