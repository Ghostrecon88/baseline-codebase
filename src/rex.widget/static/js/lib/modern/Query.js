/**
 * Bindings to HTSQL query HTTP API.
 *
 * @copyright 2015, Prometheus Research LLC
 */
'use strict';

var request   = require('../request');

class Query {

  constructor(path) {
    this.path = path;
  }

  produce(params) {
    return request('GET', this.path)
      .query(params)
      .set('Accept', 'application/json')
      .promise()
      .then(this._handleResponse);
  }

  produceCollection(params) {
    return this.produce()
  }

  produceEntity(params) {
    return this.produce()
  }
}

module.exports = Query;
