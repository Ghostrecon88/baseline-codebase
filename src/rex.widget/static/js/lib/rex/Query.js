/**
 * @jsx React.DOM
 */
'use strict';

var React = require('react');

var Query = React.createClass({

  render() {
    var {data: {data}} = this.props;
    return <div>{JSON.stringify(data)}</div>;
  }
});

module.exports = Query;
