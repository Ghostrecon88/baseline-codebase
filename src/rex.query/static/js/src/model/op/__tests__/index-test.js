import {def, pipeline, navigate, select} from '../../Query';
import * as qp from '../../QueryPointer';
import {insertAfter, removeAt} from '../index';

describe('insertAfter()', function() {

  it('individual!name', function() {
    let query = navigate('individual');
    let pointer = qp.make(query);
    let {query: nextQuery} = insertAfter({pointer, selected: null}, navigate('name'));
    expect(nextQuery).toMatchSnapshot();
  });

  it('[individual]!name', function() {
    let query = pipeline(navigate('individual'));
    let pointer = qp.make(query);
    let {query: nextQuery} = insertAfter({pointer, selected: null}, navigate('name'));
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual.sample!name', function() {
    let query = pipeline(navigate('individual'), navigate('sample'));
    let pointer = qp.make(query);
    let {query: nextQuery} = insertAfter({pointer, selected: null}, navigate('name'));
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual!sample.name', function() {
    let query = pipeline(navigate('individual'), navigate('name'));
    let pointer = qp.select(qp.make(query), ['pipeline', 0]);
    let {query: nextQuery} = insertAfter({pointer, selected: null}, navigate('sample'));
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual:define(a = sample!name)', function() {
    let query = pipeline(
      navigate('individual'),
      def('a', navigate('sample'))
    );
    let pointer = qp.select(
      qp.make(query),
      ['pipeline', 1],
      ['binding', 'query']
    );
    let {query: nextQuery} = insertAfter({pointer, selected: null}, navigate('name'));
    expect(nextQuery).toMatchSnapshot();
  });

});

describe('removeAt()', function() {

  it('individual.name!', function() {
    let query = pipeline(navigate('individual'), navigate('name'));
    let pointer = qp.select(qp.make(query), ['pipeline', 1]);
    let {query: nextQuery} = removeAt({pointer, selected: null});
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual!.name', function() {
    let query = pipeline(navigate('individual'), navigate('name'));
    let pointer = qp.select(qp.make(query), ['pipeline', 0]);
    let {query: nextQuery} = removeAt({pointer, selected: null});
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual:select(a = sample.name!)', function() {
    let query = pipeline(
      navigate('individual'),
      select({
        a: pipeline(navigate('sample'), navigate('name'))
      })
    );
    let pointer = qp.select(
      qp.make(query),
      ['pipeline', 1],
      ['select', 'a']
    );
    let {query: nextQuery} = removeAt({pointer, selected: null});
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual:select(a = [sample!])', function() {
    let query = pipeline(
      navigate('individual'),
      select({
        a: pipeline(navigate('sample')),
      })
    );
    let pointer = qp.select(
      qp.make(query),
      ['pipeline', 1],
      ['select', 'a'],
      ['pipeline', 0]
    );
    let {query: nextQuery} = removeAt({pointer, selected: null});
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual:select(a = sample.name!, b = sample)', function() {
    let query = pipeline(
      navigate('individual'),
      select({
        a: pipeline(navigate('sample'), navigate('name')),
        b: navigate('sample'),
      })
    );
    let pointer = qp.select(
      qp.make(query),
      ['pipeline', 1],
      ['select', 'a']
    );
    let {query: nextQuery} = removeAt({pointer, selected: null});
    expect(nextQuery).toMatchSnapshot();
  });

  it('individual:define(a := sample!)', function() {
    let query = pipeline(
      navigate('individual'),
      def('a', pipeline(navigate('sample'))),
    );
    let pointer = qp.select(
      qp.make(query),
      ['pipeline', 1],
      ['binding', 'query']
    );
    let {query: nextQuery} = removeAt({pointer, selected: null});
    expect(nextQuery).toMatchSnapshot();
  });

});

