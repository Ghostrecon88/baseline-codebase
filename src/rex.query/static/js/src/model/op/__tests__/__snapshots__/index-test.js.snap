exports[`insertAfter() [individual]!name 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "name",
      "regular": false,
    },
  ],
}
`;

exports[`insertAfter() individual!name 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "name",
      "regular": false,
    },
  ],
}
`;

exports[`insertAfter() individual!sample.name 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "sample",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "name",
      "regular": false,
    },
  ],
}
`;

exports[`insertAfter() individual.sample!name 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "sample",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "name",
      "regular": false,
    },
  ],
}
`;

exports[`insertAfter() individual:define(a = sample!name) 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
    Object {
      "binding": Object {
        "name": "a",
        "query": Object {
          "context": Object {
            "domain": <DOMAIN>,
            "hasInvalidType": false,
            "prev": "<PREV CONTEXT>",
            "scope": "<SCOPE [empty]>",
            "title": null,
            "type": Object {
              "card": null,
              "domain": <DOMAIN>,
              "name": "void",
            },
          },
          "name": "pipeline",
          "pipeline": Array [
            Object {
              "context": Object {
                "domain": <DOMAIN>,
                "hasInvalidType": false,
                "prev": "<PREV CONTEXT>",
                "scope": "<SCOPE [empty]>",
                "title": null,
                "type": Object {
                  "card": null,
                  "domain": <DOMAIN>,
                  "name": "void",
                },
              },
              "name": "navigate",
              "path": "sample",
              "regular": false,
            },
            Object {
              "context": Object {
                "domain": <DOMAIN>,
                "hasInvalidType": false,
                "prev": "<PREV CONTEXT>",
                "scope": "<SCOPE [empty]>",
                "title": null,
                "type": Object {
                  "card": null,
                  "domain": <DOMAIN>,
                  "name": "void",
                },
              },
              "name": "navigate",
              "path": "name",
              "regular": false,
            },
          ],
        },
      },
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "define",
    },
  ],
}
`;

exports[`removeAt() individual!.name 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "name",
      "regular": false,
    },
  ],
}
`;

exports[`removeAt() individual.name! 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
  ],
}
`;

exports[`removeAt() individual:define(a := sample!) 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
  ],
}
`;

exports[`removeAt() individual:select(a = [sample!]) 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
  ],
}
`;

exports[`removeAt() individual:select(a = sample.name!) 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
  ],
}
`;

exports[`removeAt() individual:select(a = sample.name!, b = sample) 1`] = `
Object {
  "context": Object {
    "domain": <DOMAIN>,
    "hasInvalidType": false,
    "prev": "<PREV CONTEXT>",
    "scope": "<SCOPE [empty]>",
    "title": null,
    "type": Object {
      "card": null,
      "domain": <DOMAIN>,
      "name": "void",
    },
  },
  "name": "pipeline",
  "pipeline": Array [
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "here",
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "navigate",
      "path": "individual",
      "regular": false,
    },
    Object {
      "context": Object {
        "domain": <DOMAIN>,
        "hasInvalidType": false,
        "prev": "<PREV CONTEXT>",
        "scope": "<SCOPE [empty]>",
        "title": null,
        "type": Object {
          "card": null,
          "domain": <DOMAIN>,
          "name": "void",
        },
      },
      "name": "select",
      "select": Object {
        "b": Object {
          "context": Object {
            "domain": <DOMAIN>,
            "hasInvalidType": false,
            "prev": "<PREV CONTEXT>",
            "scope": "<SCOPE [empty]>",
            "title": null,
            "type": Object {
              "card": null,
              "domain": <DOMAIN>,
              "name": "void",
            },
          },
          "name": "navigate",
          "path": "sample",
          "regular": false,
        },
      },
    },
  ],
}
`;
